org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -607,6 +623,13 @@ public class Preferences {
 		JAVA_COMPLETION_FILTERED_TYPES_DEFAULT.add("sun.*");
 
 		JAVA_RESOURCE_FILTERS_DEFAULT = Arrays.asList("node_modules", ".git");
+
+		builtinNonnullTypes.put("javax.annotation.Nonnull", "jsr305");
+		builtinNonnullTypes.put("org.eclipse.jdt.annotation.NonNull", "org.eclipse.jdt.annotation");
+		builtinNonnullTypes.put("org.springframework.lang.NonNull", "spring-core");
+		builtinNullableTypes.put("javax.annotation.Nullable", "jsr305");
+		builtinNullableTypes.put("org.eclipse.jdt.annotation.Nullable", "org.eclipse.jdt.annotation");
+		builtinNullableTypes.put("org.springframework.lang.Nullable", "spring-core");
```

Comment:
Even though you don't expose more to the default client setting, but you still can add more known null libraries to the whitelist of server side:

'io.micrometer.core.lang.Nullable' -> 'micrometer-core'
'org.jetbrains.annotations.Nullable' -> 'org.jetbrains.annotations'